samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts

```diff
@@ -0,0 +1,211 @@
+                );
+            }
+        }
+    }
+
+    private static requestSingleInstance(): void {
+        // Prevent private URI scheme notifications on Windows + Linux from creating a new instance of the application
+        const gotTheLock = app.requestSingleInstanceLock();
+        if (!gotTheLock) {
+            Main.application.quit();
+        } else {
+            Main.application.on("second-instance", Main.onSecondInstance);
+            Main.application.on("ready", Main.onReady);
+            Main.application.on("open-url", Main.onOpenUrl);
+        }
+    }
+
+    private static async loadBaseUI(): Promise<void> {
+        await Main.mainWindow.loadFile(path.join(__dirname, "../index.html"));
+    }
+
+    private static onWindowAllClosed(): void {
+        // Windows and Linux will quit the application when all windows are closed
+        if (process.platform !== "darwin") {
+            // macOS requires explicit quitting
+            Main.application.quit();
+        }
+    }
+
+    private static onClose(): void {
+        Main.mainWindow = null;
+    }
+
+    /**
+     * On Windows and Linux, this is where we receive login responses
+     */
+    private static onSecondInstance(
+        event: any,
+        commandLine: string[],
+        workingDirectory: string
+    ): void {   
+        // Someone tried to run a second instance, we should focus our window.
+        if (Main.mainWindow) {
+            if (Main.mainWindow.isMinimized()) Main.mainWindow.restore();
+            Main.mainWindow.focus();
+        }
+
+        const url = Main.getDeepLinkUrl(commandLine);
+
+        if (url) {
+            Main.getCodeFromDeepLinkUrl(url);
+        }
+    }
+
+    private static onOpenUrl(event: any, schemeData: string) {
+        event.preventDefault();
+
+        if (Main.mainWindow) {
+            if (Main.mainWindow.isMinimized()) Main.mainWindow.restore();
+            Main.mainWindow.focus();
+        }
+
+        Main.getCodeFromDeepLinkUrl(schemeData);
+    }
+
+    /**
+     * Get the deep Linked from command line params
+     */
+    private static getDeepLinkUrl(argv: string[]): string {
+        for (const arg of argv) {
+            const value = arg;
+            if (value.indexOf(authConfig.customProtocol.name) !== -1) {
+                return value;
+            }
+        }
+        return null;
+    }
+
+    private static onReady(): void {
+        Main.createMainWindow();
+        Main.mainWindow.loadFile(path.join(__dirname, "../index.html"));
+        Main.mainWindow.on("closed", Main.onClose);
+        Main.authProvider = new AuthProvider(Main.authConfig);
+        Main.fetchManager = new FetchManager();
+        Main.registerSubscriptions();
+
+        Main.attemptSSOSilent();
+    }
+
+    // Creates main application window
+    private static createMainWindow(): void {
+        this.mainWindow = new BrowserWindow({
+            width: 1000,
+            height: 1000,
```

Comment:
Should you define 1000 as a constant with an appropriate name?

Reply:
Removed magic numbers